src/ProxyClient/Varnish.php

```diff
@@ -110,7 +110,7 @@ public function banPath(string $path, ?string $contentType = null, array|string|
             if (!count($hosts)) {
                 throw new InvalidArgumentException('Either supply a list of hosts or null, but not an empty array.');
             }
-            $hosts = '^('.implode('|', $hosts).')$';
+            $hosts = '^('.implode('|', array_map('preg_quote', $hosts)).')$';
```

Comment:
Since `3.x` is PHP 8.1+, should we use a first class callable here?

```suggestion
            $hosts = '^('.implode('|', array_map(preg_quote(...), $hosts)).')$';
```

Reply:
ah, good point. i added that in #594 and also adjust the test expectation to make the test not fail